conjur/config.py

```diff
@@ -29,25 +31,29 @@ class Config():
     FIELDS = [
         ('conjur_account', 'account', True),
         ('conjur_url', 'url', True),
-        ('cert_file', 'ca_bundle', False),
+        ('cert_file', 'ca_bundle', True),
     ]
 
     _config = {}
 
     def __init__(self, config_file=DEFAULT_CONFIG_FILE):
         # pylint: disable=logging-fstring-interpolation
-        logging.debug(f"Fetching connection details from filesystem {config_file}")
 
+        logging.debug(f"Fetching connection details from filesystem {config_file}")
         config = None
         with open(config_file, 'r') as config_fp:
             config = load(config_fp, Loader=Loader)
 
-        for config_field_name, attribute_name, mandatory in self.FIELDS:
-            if mandatory:
-                assert config_field_name in config
-
-            setattr(self, attribute_name, config[config_field_name])
-            self._config[attribute_name] = getattr(self, attribute_name)
+        if config:
+            # TODO consider using load_from_file (ConjurrcData) instead of the following logic
+            # Sets the value of that FIELDS maps to with the values found in the conjurrc
+            for config_field_name, attribute_name, mandatory in self.FIELDS:
+                if mandatory and config_field_name not in config:
+                    raise InvalidConfigurationException
+                setattr(self, attribute_name, config[config_field_name])
```

Comment:
If we are going to refactor that file I think we should add a note here fo removing the setattr use.

Reply:
i added the TODO note "# TODO consider using load_from_file (ConjurrcData) instead of the following logic" so I think it is clear enough that this whole section needs to be rethought. Do you still think a comment should made here even with this comment?